jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -721,6 +766,10 @@ private static class DynamicErrorProcessor extends ErrorProcessor {}
 
     class ServerContext extends Attributes.Wrapper implements Context
     {
+        private final File jettyBase = IO.asFile(System.getProperty("jetty.base"));
+        private final File workDir = jettyBase != null && jettyBase.isDirectory() && jettyBase.canWrite() ? new File(jettyBase, "work") : null;
```

Comment:
I think this is a behaviour change. If `jetty.base` exists, then the server's tmp directory will be `$jetty.base/work`, and I assume contexts will create their own tmp dirs inside of that, but they won't be persistent unless `context.setPersistTempDirectory(true)` has been called. In previous versions of jetty, any tmp `work` directory has always been automatically persisted.

Reply:
Well it is a change because servers nor contexts ever had temporary directories before. 
But I don't think it is a change for `WebAppContext`s as the `WebInfConfiguration` also checks for the "work" name in it's parent.

Potentially `ContextHandler` could also look for a "work" parent in any temporary directory set on it and force it to be persistent if found???